epic merchant PR

Diegoflores31 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->
adds a lot of new moneysinks for the merchant, prices are a little high in order to not overshadow the Smiths or crafting roles . the average is 2X of the current player price.
Allows Merchant to buy Raw materials at a higher price than the stockpile.

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Adds new shipments for the goldface
add: Adds new categories for the goldface
refactor: Goldface UI changed size and format . its now 2 rows instead of one and will automatically acomodate new categories.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

## Pre-Merge Checklist
<!-- Don't bother filling these in while creating your Pull Request, just click the checkboxes after the Pull Request is opened and you are redirected to the page. -->
- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.
<!-- Neither the compiler nor workflow checks are perfect at detecting runtimes and errors. It is important to test your code/feature/fix locally. -->

Diegoflores31 commented:
finally

grungussuss commented on code/modules/cargo/supply_packs/tools.dm:
```diff
@@ -167,9 +146,9 @@
-	contains = /obj/item/fishing/bait/deluxe
+	name = "Fishing Grub x5"
+	cost = 20
+	contains = list(/obj/item/fishing/bait/deluxe,/obj/item/fishing/bait/deluxe,/obj/item/fishing/bait/deluxe,/obj/item/fishing/bait/deluxe,/obj/item/fishing/bait/deluxe)
```
god this sucks, we should make it an assoc of item to amount, but you don't have to

heepox commented:
I mean the ore prices are funny since you can just buy the cheapest smeltable item for the material anyway

heepox commented:
also uh the new categories should be capitalized?
could also add a category for storage for all the bags and pouches and sheathes...

grungussuss commented:
https://github.com/Monkestation/Vanderlin/issues/2817
https://github.com/Monkestation/Vanderlin/issues/2819

Diegoflores31 commented:
> #2817 #2819

fixed

heepox commented:
Right, so puffers dont have ammo being sold yet? need to add that for sure.

some items prices are not what they say they are. (sometimes i dont even spend a mammon buying some items for some reason but thats nothing new.)

still could add more categories for containers and the like.

hmm... its a good PR

Diegoflores31 commented:
> Right, so puffers dont have ammo being sold yet? need to add that for sure.
> 
> some items prices are not what they say they are. (sometimes i dont even spend a mammon buying some items for some reason but thats nothing new.)
> 
> still could add more categories for containers and the like.
> 
> hmm... its a good PR
> 
> pair it with merchant not being an absolute wet paperbag and give them some more skills so they can do stevedore stuff? ( i love my stevedores but they arent always playing.)


ammo is sold separately. you can buy it on the "ammo" section or make the artificer craft it for you

heepox commented:
also uh not all the imports like gold or silver are in the goldface?
still propably missing quite a bit...
also uh some of the stuff we can buy is old pathed somehow and is not what it's supposed to be since it does default smelts into ash. or weird smelts like steel items into iron, but thats nothing new... just an old problem.

Diegoflores31 commented:
> > ammo is sold separately. you can buy it on the "ammo" section or make the artificer craft it for you
> 
> yeah you sell the powder, but not the ammo... need to make the lead balls on the artificer table.

its there . its called "pouch of lead bullets"

heepox commented:
> > > ammo is sold separately. you can buy it on the "ammo" section or make the artificer craft it for you
> > 
> > 
> > yeah you sell the powder, but not the ammo... need to make the lead balls on the artificer table.
> 
> its there . its called "pouch of lead bullets"

no?
<img width="318" height="238" alt="image" src="https://github.com/user-attachments/assets/a7540bfd-afa3-4f58-a44b-ddedbfa4a961" />
lets test this.

Diegoflores31 commented:
> > > > ammo is sold separately. you can buy it on the "ammo" section or make the artificer craft it for you
> > > 
> > > 
> > > yeah you sell the powder, but not the ammo... need to make the lead balls on the artificer table.
> > 
> > 
> > its there . its called "pouch of lead bullets"
> 
> no? <img alt="image" width="318" height="238" src="https://private-user-images.githubusercontent.com/48312144/478609892-a7540bfd-afa3-4f58-a44b-ddedbfa4a961.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUzMDU4NjgsIm5iZiI6MTc1NTMwNTU2OCwicGF0aCI6Ii80ODMxMjE0NC80Nzg2MDk4OTItYTc1NDBiZmQtYWZhMy00ZjU4LWE0NGItZGRlZGJmYTRhOTYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE2VDAwNTI0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM0YWQxZWMxOWUyMTAzM2RhOTBjYTk1MzcwZTE0ODVkODRhYWViYzI0OWVkZTc1OTg0Y2MyMjI1MWMzN2RjNmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.lwwpGS054K6XDgNuhOy2P8n63MnZqcHwf4w-7xg-RPs"> lets test this.

its not showing because of a path issue i will fix it later

Diegoflores31 commented:
> also uh not all the imports like gold or silver are in the goldface? still propably missing quite a bit... also uh some of the stuff we can buy is old pathed somehow and is not what it's supposed to be since it does default smelts into ash. or weird smelts like steel items into iron, but thats nothing new... just an old problem.

i really dont want merchant to print his own money . thats a priviledge of stewards

grungussuss commented:
what about https://github.com/Monkestation/Vanderlin/issues/2820 ?

is that fixed?

Diegoflores31 commented:
> what about #2820 ?
> 
> is that fixed?

unrelated to the PR . likely an issue with the UI itself . will look into it.

Diegoflores31 commented on code/modules/cargo/supply_packs/weapons.dm:
```diff
@@ -112,57 +149,107 @@
+	group = "Weapons (Ranged)"
+
+/datum/supply_pack/weapons/ranged/puffer
+	name = "Smuggled Püffer"
```
can you elaborate on that ?

Diegoflores31 commented:
SOME DISCORD SUGGESTIONS APPLIED . 

NEW STORAGE TAB
SPLIT ARMOR TABS IRON/STEEL , ADDED MORE STEEL OPTIONS , TWEAKED PRICES TO MATCH THE SPREADSHEET
